java/core/src/main/java/com/whylogs/core/views/ColumnProfileView.java

```diff
@@ -0,0 +1,129 @@
+package com.whylogs.core.views;
+
+import com.whylogs.core.SummaryConfig;
+import com.whylogs.core.errors.UnsupportedError;
+import com.whylogs.core.metrics.Metric;
+import com.whylogs.core.metrics.components.MetricComponent;
+import java.util.*;
+import lombok.Getter;
+
+@Getter
+public class ColumnProfileView {
+  private HashMap<String, Metric> metrics;
+  private int successes = 0;
+  private int failures = 0;
+
+  public ColumnProfileView(HashMap<String, Metric> metrics) {
+    this.metrics = metrics;
```

Comment:
This should probably deep copy?

Reply:
There is no deep copy in java outside use the deserialziation and serialization method although Apache Commons Lang has one according to this [article](https://www.baeldung.com/java-copy-hashmap). I have changed it over to `this.metrics.pullAll(metrics)`. This does a shallow copy

Reply:
We might need to roll our own deep copy.

Reply:
kk I'm ok with that. :) So when talking through deep copy do we want each metric to also be copied into a new metric?